login/login.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package login
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/juju/cmd"
+	"github.com/juju/usso"
+	"golang.org/x/crypto/ssh/terminal"
+	"gopkg.in/errgo.v1"
+)
+
+type TokenGetter interface {
+	GetTokenWithOTP(email, password, otp, tokenName string) (*usso.SSOData, error)
+}
+
+var ussoServer (TokenGetter) = usso.ProductionUbuntuSSOServer
+
+// LoginUSSO reads login parameters from ctxt.Stdin and then retrieves an
+// oauth token from Ubuntu SSO.
+func LoginUSSO(ctx *cmd.Context, twoFactor bool, store TokenStore) (*usso.SSOData, error) {
```

Comment:
I don't think it's appropriate for this function to take a cmd.Context. This package could be useful in contexts other than command-line commands too.

How about providing a function that it can use to query the user?
For example:

    func LoginUSSO(twoFactor bool, store TokenStore, queryUser func(prompt string, secret bool) (string, error)) (*usso.SSOData, error) {

That way it's easier to test independently of I/O too.